features/device-config/service/src/main/java/org/opennms/features/deviceconfig/service/impl/DeviceConfigServiceImpl.java

```diff
@@ -177,6 +179,15 @@ private CompletableFuture<PollerResponse> pollDeviceConfig(String ipAddress, Str
         final var match = getPollerConfig().findService(ipAddress, serviceName)
                 .orElseThrow(IllegalArgumentException::new);
 
+        final Set<String> serviceNames = sessionUtils.withReadOnlyTransaction(() -> ipInterfaceDao.findByIpAddressAndLocation(ipAddress, location)
```

Comment:
May be there is a chance to consolidate DB calls in 182 and 193 since they are querying for the same thing.

Reply:
done